skiko/src/awtMain/cpp/linux/swingRedrawer.cc

```diff
@@ -0,0 +1,187 @@
+class OffScreenBuffer
+{
+public:
+    Display* display;
+    GLXPbuffer pbuffer;
+    int width;
+    int height;
+
+    OffScreenBuffer(OffScreenContext* context, int _width, int _height) {
+        int pbufferAttribs[] = {
+            GLX_PBUFFER_WIDTH, width,
+            GLX_PBUFFER_HEIGHT, height,
+            None  
+        };
+
+        width = _width;
+        height = _height;
+        display = context->display;
+
+        pbuffer = glXCreatePbuffer(context->display, context->fbConfigs[0], pbufferAttribs);
+    }
+
+    ~OffScreenBuffer() {
+        glXDestroyPbuffer(display, pbuffer);
+    }
+};
+
+class OffScreenTexture
+{
+public:
+    GLuint tex;
+    GLuint fbo;
+
+    OffScreenTexture(GLuint _tex, GLuint _fbo) {
+        tex = _tex;
+        fbo = _fbo;
+    }
+
+    ~OffScreenTexture() {
+        glDeleteFramebuffers(1, &fbo);
+        glDeleteTextures(1, &tex);
+    }
+};
+
+extern "C"
+{
+    JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_swing_LinuxOpenGLSwingRedrawer_makeOffScreenContext(
+        JNIEnv *env, jobject redrawer)
+    {
+        OffScreenContext* context = new OffScreenContext();
+        return toJavaPointer(context);
+    }
+
+    JNIEXPORT void JNICALL Java_org_jetbrains_skiko_swing_LinuxOpenGLSwingRedrawer_disposeOffScreenContext(
+        JNIEnv *env, jobject redrawer, jlong contextPtr)
+    {
+        OffScreenContext* context = fromJavaPointer<OffScreenContext *>(contextPtr);
+        delete context;
+    }
+
+    JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_swing_LinuxOpenGLSwingRedrawer_makeOffScreenBuffer(
+        JNIEnv *env, jobject redrawer, jlong contextPtr, jlong oldBufferPtr, jint width, jint height)
+    {
+        OffScreenContext* context = fromJavaPointer<OffScreenContext *>(contextPtr);
+        OffScreenBuffer* oldBuffer = fromJavaPointer<OffScreenBuffer *>(oldBufferPtr);
+        
+        OffScreenBuffer* buffer;
+        if (oldBuffer == nullptr || oldBuffer->width != width || oldBuffer->height != height) {
+            if (oldBuffer != nullptr) {
+                delete oldBuffer;
+            }
+            buffer = new OffScreenBuffer(context, width, height);
+        } else {
+            buffer = oldBuffer;
+        }
+
+        return toJavaPointer(buffer);
+    }
+
+    JNIEXPORT void JNICALL Java_org_jetbrains_skiko_swing_LinuxOpenGLSwingRedrawer_disposeOffScreenBuffer(
+        JNIEnv *env, jobject redrawer, jlong bufferPtr)
+    {
+        OffScreenBuffer* buffer = fromJavaPointer<OffScreenBuffer *>(bufferPtr);
+        delete buffer;
+    }
+
+    JNIEXPORT void JNICALL Java_org_jetbrains_skiko_swing_LinuxOpenGLSwingRedrawer_startRendering(
+        JNIEnv *env, jobject redrawer, jlong contextPtr, jlong bufferPtr)
+    {
+        OffScreenContext* context = fromJavaPointer<OffScreenContext *>(contextPtr);
+        OffScreenBuffer* buffer = fromJavaPointer<OffScreenBuffer *>(bufferPtr);
+
+        glXMakeCurrent(context->display, buffer->pbuffer, context->context);
```

Comment:
[Similarly, can fail](https://registry.khronos.org/OpenGL-Refpages/gl2.1/xhtml/glXMakeCurrent.xml)
Did you test it on two screens?

Reply:
This error appear even with one screen but on window resize